js/components/src/components/mobile-player/rotation-lock.tsx

```diff
@@ -0,0 +1,162 @@
+import * as ScreenOrientation from "expo-screen-orientation";
```

Comment:
Anything that's a new native dependency in @components we need to make optional, I think, we can't force our users to definitely 100% have this installed during an OTA update

I think all you gotta do is wrap this in a async loader (like video-async.native.tsx) and then use require() to check if the module is installed https://github.com/streamplace/streamplace/blob/next/js/components/src/crypto-polyfill.native.tsx

Actually you might have to figure out how to native module detection, as require() will just tell you if the JS is there